enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,924 @@
+* There could be potentially large version skew between the management cluster and the guest cluster.
+  * Client-go used by the operators may be either older or newer than the API server that runs the management cluster.
+    We expect that the API and protocols to access it remains stable and such older / newer client-go can access it.
+  * CI jobs that check that were suggested above in E2E test chapter. 
+* Upstream components (CSI driver, csi-snapshot-controller) will always get kubeconfig only to the guest
+  cluster, so only the same version skew as in standalone OCP can occur there.
+  
+<!--
+How will the component handle version skew with other components?
+What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI
+  or CNI may require updating that component before the kubelet.
+-->
+
+### Operational Aspects of API Extensions
+
+N/A? This enhancement does not add new APIs nor its extensions.
+
+<!--
+Describe the impact of API extensions (mentioned in the proposal section, i.e. CRDs,
+admission and conversion webhooks, aggregated API servers, finalizers) here in detail,
+especially how they impact the OCP system architecture and operational aspects.
+
+- For conversion/admission webhooks and aggregated apiservers: what are the SLIs (Service Level
+  Indicators) an administrator or support can use to determine the health of the API extensions
+
+  Examples (metrics, alerts, operator conditions)
+  - authentication-operator condition `APIServerDegraded=False`
+  - authentication-operator condition `APIServerAvailable=True`
+  - openshift-authentication/oauth-apiserver deployment and pods health
+
+- What impact do these API extensions have on existing SLIs (e.g. scalability, API throughput,
+  API availability)
+
+  Examples:
+  - Adds 1s to every pod update in the system, slowing down pod scheduling by 5s on average.
+  - Fails creation of ConfigMap in the system when the webhook is not available.
+  - Adds a dependency on the SDN service network for all resources, risking API availability in case
+    of SDN issues.
+  - Expected use-cases require less than 1000 instances of the CRD, not impacting
+    general API throughput.
+
+- How is the impact on existing SLIs to be measured and when (e.g. every release by QE, or
+  automatically in CI) and by whom (e.g. perf team; name the responsible person and let them review
+  this enhancement)
+-->
+
+#### Failure Modes
+
+Same as in standalone OCP - status of all storage operators is reported to ClusterOperator CRs in the guest cluster.
```

Comment:
One new failure mode that we haven't really discussed yet: what happens if the management cluster loses connection or access to the guest cluster?
This isn't specific to storage, but it's a new failure mode with hypershift in general. Maybe that's documented elsewhere (or should be).